### DISCUSSION TITLE
Can default <C-hjkl> keymaps be removed without disabling all default keymaps?

### DISCUSSION THREAD

**Discussion by ooloth** [LazyVim/LazyVim#277] [Q&A] [answered]
Loving LazyVim!

Unfortunately, these four default keymaps [interfere with the vim-tmux-navigator keymaps for moving from vim to tmux windows](https://github.com/christoomey/vim-tmux-navigator#vim---tmux-doesnt-work):

```lua
-- Move to window using the <ctrl> hjkl keys
map("n", "<C-h>", "<C-w>h", { desc = "Go to left window" })
map("n", "<C-j>", "<C-w>j", { desc = "Go to lower window" })
map("n", "<C-k>", "<C-w>k", { desc = "Go to upper window" })
map("n", "<C-l>", "<C-w>l", { desc = "Go to right window" })
```

The solution is for me to remove these keymaps -- is there a way to do that without disabling the whole file? The other keymaps are useful and I'd like to benefit from any that may be added in the future.

(Another potential solution would be for LazyVim to replace these four keymaps with [christoomey/vim-tmux-navigator](https://github.com/christoomey/vim-tmux-navigator) since it provides the same functionality for `vim<->vim` window navigation with the bonus of enabling it for `vim<->tmux` window navigation as well.)

**Comment by dpetka2001** [comment]
Can you not just try to change them to do what you want in `keymaps.lua`? Alternatively you can just delete them and then set your new keys in your plugin's configuration. I tried for example
```lua
del = vim.keymap.del
del("n", "<C-k>")
```
in `keymaps.lua` and does indeed remove the mapping.

**Comment by folke** [comment, marked as answer]
Just add the **vim-tmux-navigator** mappings in your `lua/config/keymaps.lua` file and it should work. No need to delete the existing mappings, since the new mappings will just overwrite the existing ones.

**Reply by ooloth** [reply]
That worked, thanks!

I added the following to `lua/config/keymaps.lua`:

```lua
-- overwrite lazyvim mappings with vim-tmux-navigator mappings
-- see: https://github.com/christoomey/vim-tmux-navigator/blob/master/plugin/tmux_navigator.vim
vim.cmd([[
  noremap <silent> <c-h> :<C-U>TmuxNavigateLeft<cr>
  noremap <silent> <c-j> :<C-U>TmuxNavigateDown<cr>
  noremap <silent> <c-k> :<C-U>TmuxNavigateUp<cr>
  noremap <silent> <c-l> :<C-U>TmuxNavigateRight<cr>
  noremap <silent> <c-\> :<C-U>TmuxNavigatePrevious<cr>
]])
```

**Reply by sdondley** [reply]
I tried this suggestion as well. It did not work for me.

**Reply by epinadev** [reply]
worked like a chamr

**Comment by hasecilu** [comment]
This is the overwrite using the `map` function from `lua/config/keymaps.lua`
```lua
-- vim-tmux-navigator
map({ "i", "n", "v" }, "<C-k>", "<cmd>TmuxNavigateUp<cr><esc>", { desc = "Move cursor to top pane" })
map({ "i", "n", "v" }, "<C-j>", "<cmd>TmuxNavigateDown<cr><esc>", { desc = "Move cursor to bottom pane" })
map({ "i", "n", "v" }, "<C-h>", "<cmd>TmuxNavigateLeft<cr><esc>", { desc = "Move cursor to left pane" })
map({ "i", "n", "v" }, "<C-l>", "<cmd>TmuxNavigateRight<cr><esc>", { desc = "Move cursor to right pane" })
map({ "i", "n", "v" }, "<C-\\>", "<cmd>TmuxNavigatePrevious<cr><esc>", { desc = "Move cursor to previous pane" })
```

**Reply by sdondley** [reply]
This doesn't work for me. Each of those lines say "Undefined global `map`

**Reply by dpetka2001** [reply]
You get "undefined global map" because that is a shortcut used by the user. Try to set the keymaps using `vim.keymap.set` instead of `map`.

**Reply by hasecilu** [reply]
Here is the definition of the `map` function: https://github.com/LazyVim/LazyVim/blob/main/lua/lazyvim/config/keymaps.lua

You'll need to copy the function to your `lua/config/keymaps.lua` file.

**Comment by sdondley** [comment]
I can't seem to get things working with the ctrl keys. I got this in keymaps.lua

```
vim.cmd([[
 let g:tmux_navigator_no_mappings = 1
 let g:tmux_navigator_disable_when_zoomed = 0
  noremap <silent> <c-h> :<C-U>TmuxNavigateLeft<cr>
    noremap <silent> <c-j> :<C-U>TmuxNavigateDown<cr>
    noremap <silent> <leader>y :<C-U>TmuxNavigateUp<cr>
    noremap <silent> <c-l> :<C-U>TmuxNavigateRight<cr>
    noremap <silent> <c-\> :<C-U>TmuxNavigatePrevious<cr>
  ]])
```

`<leader>y` works perfectly fine so I know these keymaps are seen by lazy vim. But `<c-h,j,l>` are ignored.

**Reply by abeldekat** [reply]
Hello @sdondley,
Just for completeness: 
I noticed you found the solution, described in this [vim-tmux-navigator issue](https://github.com/christoomey/vim-tmux-navigator/issues/359)
